### PR TITLE
Event logger disable

### PR DESCRIPTION
**Pull request by lucasmoura**
## Proposed Commit Message
event logger disable

Adds json response to disable operation.

## Test Steps
Run the modified integration tests

## Desired commit type

 - [ ] Squash and merge: Using the "Proposed Commit Message"
 - [ ] Create a merge commit and use "Proposed Commit Message"
 - [x] Rebase and merge, no merge commit, rely only on existing commit messages

## Checklist

 - [x] I have updated or added any unit tests accordingly
 - [x] I have updated or added any integration tests accordingly
 - [ ] I have updated or added any documentation accordingly

### CODE REVIEW

File: uaclient/entitlements/base.py [hunk 1]
```diff
@@ -665,30 +717,49 @@ def _check_for_reboot_msg(self, operation: str) -> None:
                 )
             )
 
-    def disable(self, silent: bool = False) -> bool:
+    def disable(
+        self, silent: bool = False
+    ) -> Tuple[bool, Union[None, CanDisableFailure]]:
```
Comment (orndorffgrant) [inline uaclient/entitlements/base.py:722]: I believe `Union[None, X]` is equivalent to `Optional[X]`

File: uaclient/entitlements/tests/test_base.py [hunk 2]
```diff
@@ -116,22 +120,72 @@ def test_init_accepts_a_uaconfig(self):
...
             "Test Concrete Entitlement is not currently enabled\n"
-            "See: sudo ua status\n"
+            "See: sudo ua status"
         )
-        stdout, _ = capsys.readouterr()
-        assert expected_stdout == stdout
+        assert expected_msg == fail.message
+
+    @mock.patch("uaclient.entitlements.entitlement_factory")
+    def test_can_disable_false_on_depedent_service(
```
Comment (orndorffgrant) [inline uaclient/entitlements/tests/test_base.py:141]: Small typo
```suggestion
    def test_can_disable_false_on_dependent_service(
```

File: uaclient/entitlements/tests/test_base.py [hunk 3]
```diff
@@ -116,22 +120,72 @@ def test_init_accepts_a_uaconfig(self):
...
+        ret, fail = entitlement.can_disable()
+        assert not ret
+        assert (
+            fail.reason
+            == status.CanDisableFailureReason.ACTIVE_DEPENDENT_SERVICES
+        )
+        assert fail.message is None
+
+    @mock.patch("uaclient.entitlements.entitlement_factory")
+    def test_can_disable_when_ignoring_depedent_service(
```
Comment (orndorffgrant) [inline uaclient/entitlements/tests/test_base.py:168]: same here
```suggestion
    def test_can_disable_when_ignoring_dependent_service(
```

**Review by orndorffgrant — COMMENTED** [review]
Code looks good! Just a couple small things. I'll run the @slow tests and play with it too

File: features/attached_commands.feature [hunk 4]
```diff
@@ -121,6 +121,49 @@ Feature: Command behaviour when attached to an UA subscription
            | impish  |
            | jammy   |
 
+    @series.lts
+    @uses.config.machine_type.lxd.container
+    Scenario Outline: Attached disable of an already disabled service in a ubuntu machine
+        Given a `<release>` machine with ubuntu-advantage-tools installed
+        When I attach `contract_token` with sudo
+        Then I verify that running `ua disable foobar --format json` `as non-root` exits `1`
```
[the remark this answers is not in the document]
Comment (lucasmoura) [reply features/attached_commands.feature:129]: Yes, we definitely need that. I think I lost that work during the rebase operation :facepalm: I will update the code

**Comment by lucasmoura** [comment]
@orndorffgrant code updated

**Review by orndorffgrant — APPROVED** [review]
Latest changes look good to me! Let's just wait on Jenkins

**Comment by lucasmoura** [comment]
@orndorffgrant jenkins errors seem unrelated to the proposed changes

**Comment by orndorffgrant** [comment]
They do seem like lxd flakiness. Unfortunately, two of the tests that suffered were directly related to this PR :(

I'll just run them locally to double check instead of retriggering jenkins

**Comment by orndorffgrant** [comment]
Everything passed for me! Merging now

**Pull request merged**